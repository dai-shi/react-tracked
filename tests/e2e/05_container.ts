import { describe, expect, it } from 'vitest';
import puppeteer from 'puppeteer';

describe('05_container', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    // eslint-disable-next-line import/no-named-as-default-member
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(5) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(5) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(5) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(5) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(5) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(5) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(6) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(6) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(6) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(6) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(6) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(6) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(6) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await browser.close();
  });
});
