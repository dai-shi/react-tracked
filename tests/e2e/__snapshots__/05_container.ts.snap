// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`05_container > should work with recorded events 1`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 30<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 32<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 2`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 3`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 46<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 48<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 4`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 54<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 56<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 5`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 62<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 64<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 6`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 70<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 72<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 7`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 78<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 80<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 8`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 86<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 88<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 9`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 94<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 96<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 10`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 11`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 110<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 112<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 12`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 118<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 120<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 13`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 126<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 128<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 14`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 134<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 136<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 15`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 142<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 144<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 16`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 150<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 152<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 17`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 158<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 160<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 18`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 166<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 168<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 19`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 174<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 176<div><span>Count: -1</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;

exports[`05_container > should work with recorded events 20`] = `
"
    <div id="app"><h1>First container</h1><div>numRendered: 102<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 104<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Second container</h1><div>numRendered: 182<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 184<div><span>Count: -2</span><button type="button">+1</button><button type="button">-1</button></div></div></div>
  

"
`;
